python/ray/util/collective/collective.py

```diff
@@ -0,0 +1,261 @@
+"""APIs exposed under the namespace ray.util.collective."""
+import logging
+
+import ray
+from ray.util.collective import types
+from ray.util.collective.const import get_nccl_store_name
+
+# Get the availability information first by importing information
+_MPI_AVAILABLE = False
+_NCCL_AVAILABLE = True
+
+# try:
+#     from ray.util.collective.collective_group.mpi_collective_group
+#     import MPIGroup
+# except ImportError:
+#     _MPI_AVAILABLE = False
+
+try:
+    from ray.util.collective.collective_group import NCCLGroup
+except ImportError:
+    _NCCL_AVAILABLE = False
+
+logger = logging.getLogger(__name__)
+
+
+def nccl_available():
+    return _NCCL_AVAILABLE
+
+
+def mpi_available():
+    return _MPI_AVAILABLE
+
+
+class GroupManager(object):
```

Comment:
This should only be created on an actor right? 

Would it be possible to raise ValueError if not on actor?